internal/driver/daemon.go

```diff
@@ -84,6 +84,49 @@ func (r *RegistryDefault) ServeWrite(ctx context.Context) func() error {
 	}
 }
 
+func (r *RegistryDefault) ServeMetrics(ctx context.Context) func() error {
+	return func() error {
+		l, err := net.Listen("tcp", r.Config().MetricsListenOn())
+		if err != nil {
+			return err
+		}
+
+		eg := &errgroup.Group{}
+		ctx, cancel := context.WithCancel(ctx)
+
+		s := graceful.WithDefaults(&http.Server{
+			Handler: r.MetricsRouter(),
+			Addr:    r.Config().MetricsListenOn(),
+		})
+
+		eg.Go(func() error {
+			if err := s.Serve(l); !errors.Is(err, http.ErrServerClosed) {
+				// unexpected error
+				return errors.WithStack(err)
+			}
+			return nil
+		})
+		eg.Go(func() error {
+			err := s.Serve(l)
+			if err != nil && !strings.Contains(err.Error(), "use of closed network connection") {
+				// unexpected error
+				return errors.WithStack(err)
+			}
+			// trigger further shutdown
+			cancel()
+			return nil
+		})
+
+		eg.Go(func() error {
+			<-ctx.Done()
+			ctx, cancel := context.WithTimeout(context.Background(), graceful.DefaultReadTimeout)
+			defer cancel()
+			return s.Shutdown(ctx)
+		})
```

Comment:
Here we only have one server, so you don't have to coordinate shutdown in case it errors. In the other case we start two servers and the multiplex listener, so it gets more complex. Here it is enough to just do

```suggestion
		eg.Go(func() error {
			retrun graceful.Graceful(s.ListenAndServe, s.Shutdown)
		})
		eg.Go(func() error {
			<-ctx.Done()
			ctx, cancel := context.WithTimeout(context.Background(), graceful.DefaultReadTimeout)
			defer cancel()
			return s.Shutdown(ctx)
		})
		
		return eg.Wait()
```